### PR TITLE
docs: new split docker hub images

### PR DESCRIPTION
**Pull request by mattklein123**

### CODE REVIEW

**Comment by mattklein123** [comment]
cc @bndw

File: docs/root/install/building.rst [hunk 1]
```diff
@@ -32,7 +32,20 @@ Pre-built binaries
...
+binary.
+
+* `envoyproxy/envoy-dev <https://hub.docker.com/r/envoyproxy/envoy-dev/tags/>`_: Release binary with
+  symbols stripped on top of an Ubuntu Xenial base.
+* `envoyproxy/envoy-alpine-dev <https://hub.docker.com/r/envoyproxy/envoy-alpine-dev/tags/>`_: Release
+  binary with symbols stripped on top of a **glibc** alpine base.
+* `envoyproxy/envoy-alpine-debug-dev <https://hub.docker.com/r/envoyproxy/envoy-alpine-debug-dev/tags/>`_:
+  Release binary with debug symbols on top of a **glibc** alpine base.
+
+In the above *dev* repositories, the *latest* tag points to last Envoy SHA in master that passed
```
Comment (htuch) [inline docs/root/install/building.rst]: Nit: "the last Envoy SHA"

File: docs/root/install/building.rst [hunk 2]
```diff
@@ -32,7 +32,20 @@ Pre-built binaries
 ------------------
 
 On every master commit we create a set of lightweight Docker images that contain the Envoy
-binary. We also tag the docker images with release versions when we do official releases.
+binary.
+
+* `envoyproxy/envoy-dev <https://hub.docker.com/r/envoyproxy/envoy-dev/tags/>`_: Release binary with
+  symbols stripped on top of an Ubuntu Xenial base.
```
Comment (htuch) [inline docs/root/install/building.rst]: Can you remind me why we have stripped binaries for some Docker images and not for others?

Comment (mattklein123) [reply docs/root/install/building.rst]: Size reasons.

File: docs/root/install/building.rst [hunk 3]
```diff
@@ -32,7 +32,20 @@ Pre-built binaries
...
+* `envoyproxy/envoy-alpine-dev <https://hub.docker.com/r/envoyproxy/envoy-alpine-dev/tags/>`_: Release
+  binary with symbols stripped on top of a **glibc** alpine base.
+* `envoyproxy/envoy-alpine-debug-dev <https://hub.docker.com/r/envoyproxy/envoy-alpine-debug-dev/tags/>`_:
+  Release binary with debug symbols on top of a **glibc** alpine base.
+
+In the above *dev* repositories, the *latest* tag points to the last Envoy SHA in master that passed
+tests.
+
+We also tag the docker images with release versions when we do official releases. These images can
```
Comment (bndw) [inline docs/root/install/building.rst]: May consider documenting the `envoyproxy/envoy` images first, if the intended audience is users and assuming most users want the official release.

**Comment by bndw** [comment]
LGTM

**Comment by mattklein123** [comment]
@bndw @htuch updated per comments.

**Pull request merged**